suite-native/passphrase/src/PassphraseFormModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useRef } from 'react';
+import { Modal, TextInput } from 'react-native';
+
+import { useSetAtom } from 'jotai';
+
+import { Form, TextInputField, useForm } from '@suite-native/forms';
+import { yup } from '@trezor/validation';
+import { Box, Button, VStack } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useTranslate } from '@suite-native/intl';
+
+import { isPassphraseModalVisibleAtom } from './passphraseModalVisibilityAtom';
+
+const PASSPHRASE_MIN_LENGTH = 1;
+const PASSPHRASE_MAX_LENGTH = 50;
+
+const passphraseFormSchema = yup.object({
+    passphrase: yup
+        .string()
+        .required('Empty passphrase.')
+        .max(PASSPHRASE_MAX_LENGTH)
+        .min(PASSPHRASE_MIN_LENGTH),
+});
+
+type PassphraseFormValues = yup.InferType<typeof passphraseFormSchema>;
+
+const modalBackgroundOverlayStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    backgroundColor: utils.transparentize(0.3, utils.colors.backgroundNeutralBold),
+}));
+
+const formStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    borderBottomLeftRadius: utils.borders.radii.large,
+    borderBottomRightRadius: utils.borders.radii.large,
+}));
+
+export const PassphraseFormModal = () => {
+    const { translate } = useTranslate();
+
+    const passphraseInputRef = useRef<TextInput>(null);
+
+    const setPassphraseModalVisibility = useSetAtom(isPassphraseModalVisibleAtom);
```

Comment:
nit: for consistency with other atom setters I would preffer:
```suggestion
    const setIsPassphraseModalVisible = useSetAtom(isPassphraseModalVisibleAtom);
```

Reply:
Makes more sense to me like this. I don't think this strictly needs to be this consistent. If it should, I would refactor other atoms, because this name makes more sense, but that's out of scope for this.

Reply:
https://github.com/trezor/trezor-suite/pull/10703/commits/8858845e04176767d6dc143dddbb6dbb73bca73c